Composer/plugins/runtimes/src/index.ts

```diff
@@ -0,0 +1,197 @@
+      const remoteBotPath = path.join(publishFolder, 'ComposerDialogs');
+      const localBotPath = path.join(runtimePath, 'ComposerDialogs');
+      await fs.copy(localBotPath, remoteBotPath, {
+        overwrite: true,
+        recursive: true,
+      });
+
+      // write settings to disk in the appropriate location
+      const settingsPath = path.join(publishFolder, 'ComposerDialogs', 'settings', 'appsettings.json');
+      // Set the bot and root fields to `ComposerDialogs` - this points the runtime to the appropriate deployed location
+      // todo: are both necessary?
+      Object.assign(settings, { bot: 'ComposerDialogs', root: 'ComposerDialogs' });
+      if (!(await fs.pathExists(path.dirname(settingsPath)))) {
+        fs.mkdirSync(path.dirname(settingsPath), { recursive: true });
+      }
+      fs.writeFileSync(settingsPath, JSON.stringify(settings, null, 2));
+
+      // return the location of the build artifiacts
+      return publishFolder;
+    },
+    eject: async (project, localDisk: IFileStorage) => {
+      const sourcePath = path.resolve(__dirname, '../../../../runtime/dotnet');
+      const destPath = path.join(project.dir, 'runtime');
+      if (!(await project.fileStorage.exists(destPath))) {
+        // used to read bot project template from source (bundled in plugin)
+        await copyDir(sourcePath, localDisk, destPath, project.fileStorage);
+        const schemaDstPath = path.join(project.dir, 'schemas');
+        const schemaSrcPath = path.join(sourcePath, 'azurewebapp/schemas');
+        const customSchemaExists = fs.existsSync(schemaDstPath);
+        const pathsToExclude: Set<string> = new Set();
+        if (customSchemaExists) {
+          const sdkExcludePath = await localDisk.glob('sdk.schema', schemaSrcPath);
+          if (sdkExcludePath.length > 0) {
+            pathsToExclude.add(path.join(schemaSrcPath, sdkExcludePath[0]));
+          }
+        }
+        await copyDir(schemaSrcPath, localDisk, schemaDstPath, project.fileStorage, pathsToExclude);
+        const schemaFolderInRuntime = path.join(destPath, 'azurewebapp/schemas');
+        await removeDirAndFiles(schemaFolderInRuntime);
+        return destPath;
+      }
+      throw new Error(`Runtime already exists at ${destPath}`);
+    },
+  });
+
+  composer.addRuntimeTemplate({
+    key: 'node-azurewebapp',
+    name: 'JS',
+    startCommand: 'node ./lib/webapp.js',
+    path: path.resolve(__dirname, '../../../../runtime/node'),
+    build: async (runtimePath: string, _project: any) => {
+      // do stuff
+      composer.log('BUILD THIS JS PROJECT');
+      const { installOut, installErr } = await exec('npm install', {
+        cwd: runtimePath,
+        stdio: 'pipe',
+      });
+      const { install2Out, install2Err } = await exec('npm build', {
+        cwd: runtimePath,
+        stdio: 'pipe',
+      });
+      composer.log('BUILD COMPLETE');
+    },
+    run: async (project: any, localDisk: IFileStorage) => {
+      // do stuff
```

Comment:
this function implementation is empty?

Reply:
I still need to move the implementation of the run into this plugin, but I wanted to get it into the code while I was updating the form of the API.  I will execute on this ASAP.